src-self-hosted/Module.zig

```diff
@@ -2072,6 +2080,15 @@ fn createNewDecl(
     return new_decl;
 }
 
+/// Get error value for error tag `name`.
+pub fn getErrorValue(self: *Module, name: []const u8) !u16 {
+    const new_val = @intCast(u16, self.global_error_set.items().len);
+    if (self.global_error_set.get(name)) |some| return some;
+
+    try self.global_error_set.put(self.gpa, try self.gpa.dupe(u8, name), new_val);
```

Comment:
This is a memory leak if the 2nd `try` here fails. I recommend to do `self.global_error_set.ensureCapacity` above, and then this can be `putAssumeCapacity`.

Looking even more closely, I think this can be optimized further with getOrPut like this:

```zig
const gop = try self.global_error_set.getOrPut(name);
if (gop.found_existing)
    return gop.entry.value;
errdefer self.global_error_set.removeAssertDiscard(name);

gop.entry = .{
    .key = try self.gpa.dupe(u8, name),
    .value = @intCast(u16, self.global_error_set.items().len),
};
```

Not bad eh? :)